product_supplierinfo_for_customer/views/product_views.xml

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2015 AvanzOSC
+     Copyright 2015-18 Tecnativa
+     Copyright 2017-19 ForgeFlow
+     Copyright 2019 Tecnativa - Pedro M. Baeza
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="product_customerinfo_form_view" model="ir.ui.view">
+        <field name="name">product.customerinfo.form.view</field>
+        <field name="model">product.customerinfo</field>
+        <field name="arch" type="xml">
+            <form string="Customer Information">
+                <group>
+                    <group string="Customer">
+                        <field
+                            name="product_id"
+                            groups="product.group_product_variant,base.group_multi_company"
+                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
+                            options="{'no_create_edit': True}"
+                        />
+                        <field
+                            name="partner_id"
+                            string="Customer"
+                            context="{'res_partner_search_mode': 'customer'}"
+                        />
+                        <field name="product_name" string="Customer Product Name" />
+                        <field name="product_code" string="Customer Product Code" />
+                        <field
+                            name="product_id"
+                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
+                            groups="product.group_product_variant,base.group_multi_company"
+                            invisible="1"
+                        />
+                    </group>
```

Comment:
```suggestion
                    <group string="Customer">
                        <field name="company_id" invisible="1"/>
                        <field
                            name="product_id"
                            groups="product.group_product_variant"
                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
                            options="{'no_create_edit': True}"
                        />
                        <field
                            name="partner_id"
                            string="Customer"
                            context="{'res_partner_search_mode': 'customer'}"
                        />
                        <field name="product_name" string="Customer Product Name" />
                        <field name="product_code" string="Customer Product Code" />
                        <field
                            name="product_id"
                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
                            groups="product.group_product_variant"
                            invisible="1"
                        />
                    </group>
```
As I said before in another suggestion, This changes solve your problem. Please, fix it.